docker_images/ci-functional/Dockerfile

```diff
@@ -67,23 +43,56 @@ RUN apt-get -qq update \
        g++-9-multilib g++-7-multilib \
     #
     # Alternatives for GCC
-    && chmod +x /root/select_gcc.sh \
     && update-alternatives --install /usr/bin/gcc gcc $GCC_9 10 \
     && update-alternatives --install /usr/bin/gcc gcc $GCC_7 20 \
     && update-alternatives --install /usr/bin/g++ g++ $GXX_9 10 \
     && update-alternatives --install /usr/bin/g++ g++ $GXX_7 20 \
     #
     # Alternatives for Clang
-    && chmod +x /root/select_clang.sh \
     && update-alternatives --install /usr/bin/clang clang $CLANG_9 10 \
     && update-alternatives --install /usr/bin/clang clang $CLANG_7 20 \
     && update-alternatives --install /usr/bin/clang++ clang++ $CLANGXX_9 10 \
     && update-alternatives --install /usr/bin/clang++ clang++ $CLANGXX_7 20 \
-    #
-    && chmod +x /root/select_alternatives.sh \
-    #
     # Clean the image
-    && rm -rf /var/lib/apt/lists
+    && rm -rf /var/lib/apt/lists && rm -rf /root/*.tar.gz
+
+RUN useradd -ms /bin/bash conan && echo "conan:conan" | chpasswd && adduser conan sudo
+RUN echo "ALL ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+USER conan
+WORKDIR /home/conan
+ENV PATH /home/conan/.pyenv/shims:/home/conan/.pyenv/bin:$PATH
+RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash \
+    && pyenv init - \
+    && pyenv virtualenv-init - \
+    && pyenv update \
+    && pyenv install 3.8.1 \
+    && pyenv install 3.7.6 \
+    && pyenv install 3.5.9 \
+    && pyenv install 2.7.17
+
+COPY --chown=conan \
+     select_cmake.sh \
+     select_gcc.sh \
+     select_clang.sh \
+     select_alternatives.sh \
+     /home/conan/
+
+RUN chmod +x /home/conan/*.sh
+RUN sudo /home/conan/select_alternatives.sh gcc 9 clang 9 cmake 3.16.3
+
 
-ENTRYPOINT ["/root/select_alternatives.sh" ]
-CMD ["gcc", "9", "clang", "9", "cmake", "3.16.4"]
+# To run the tests
+# Launch the container:
+#   docker run -it --rm IMAGENAME -vPATH_WITH_CONAN_SOURCES:/home/conan/sources /bin/bash
+# Select the python version:
+#   pyenv global 3.8.1
+# Upgrade pip (recommended):
+#    pip install --upgrade pip
+# Install requirements:
+#    pip install -r source/conans/requirements.txt
+#    pip install -r source/conans/requirements_dev.txt
+#    pip install -r source/conans/requirements_server.txt
+# Prepare PYTHONPATH:
+#   export PYTHONPATH=$(pwd)
+# Run pytest:
+#   cd source && pytest
```

Comment:
About the entrypoint, there is an issue. I cannot update the alternatives without "sudo" but if I introduce "sudo" in the entrypoint the user changes to root ?¿ I don't know if this can be solved.

Reply:
Ok, moving the "sudo" inside the script makes sense and seems to work. Will keep trying to make it work.